adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java

```diff
@@ -258,6 +262,34 @@ void removeTokenCacheItem(final TokenCacheItem tokenCacheItem, final String reso
         Telemetry.getInstance().stopEvent(mTelemetryRequestId, cacheEvent,
                 EventStrings.TOKEN_CACHE_DELETE);
     }
+
+    boolean isMultipleRTsMatchingGivenAppAndResource(final String clientId, final String resource) {
+        final Iterator<TokenCacheItem> allItems = mTokenCacheStore.getAll();
+        final List<TokenCacheItem> regularRTsMatchingRequest = new ArrayList<>();
+        while (allItems.hasNext()) {
+            final TokenCacheItem tokenCacheItem = allItems.next();
+            if (tokenCacheItem.getAuthority().equalsIgnoreCase(mAuthority) && tokenCacheItem.getClientId().equalsIgnoreCase(clientId)
+                    && resource.equalsIgnoreCase(tokenCacheItem.getResource()) && !tokenCacheItem.getIsMultiResourceRefreshToken()) {
+                regularRTsMatchingRequest.add(tokenCacheItem);
+            }
+        }
+
+        return regularRTsMatchingRequest.size() > 1;
+    }
+
+    boolean isMultipleMRRTsMatchingGivenApp(final String clientId) {
```

Comment:
>isMultipleMRRTsMatchingGivenApp [](start = 12, length = 31)

following making throwif* function pacakage scope you should make these 2 private.